### PR TITLE
Feature/ngram

### PR DESCRIPTION
**Pull request by seme0021**
Adding NGram MLeap transformer.

### CODE REVIEW

File: mleap-core/src/test/scala/ml/combust/mleap/core/feature/NGramModelSpec.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import org.scalatest.FunSpec
+
+/**
+  * Created by mikhail on 10/16/16.
+  */
+class NGramModelSpec extends FunSpec{
+  describe("#apply") {
+    it("returns an array of ngrams") {
+      val ngram = NGramModel(2)
+
```
Comment (hollinwilkins) [inline mleap-core/src/test/scala/ml/combust/mleap/core/feature/NGramModelSpec.scala]: Get rid of these new lines between the setup steps.

File: mleap-runtime/src/main/scala/ml/combust/mleap/runtime/transformer/feature/NGram.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import scala.util.Try
+
+/**
+  * Created by mikhail on 10/16/16.
+  */
+case class NGram(uid: String = Transformer.uniqueName("ngram"),
+                 inputCol: String,
+                 outputCol: String,
+                 model: NGramModel
+                ) extends Transformer{
```
Comment (hollinwilkins) [inline mleap-runtime/src/main/scala/ml/combust/mleap/runtime/transformer/feature/NGram.scala]: Get rid of the newline before this closing paren

File: mleap-runtime/src/test/scala/ml/combust/mleap/runtime/transformer/feature/NGramSpec.scala [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+      println(data(0).getArray(1)(0))
+      println(data(0).getArray(1)(1))
+
+      assert(data(0).getArray[String](1)(0) == "a b")
+      assert(data(0).getArray[String](1)(1) == "b c")
+    }
+  }
+
+  describe("with invalid input column") {
```
Comment (mishkinf) [inline mleap-runtime/src/test/scala/ml/combust/mleap/runtime/transformer/feature/NGramSpec.scala]: what about a spec with an invalid col?

File: mleap-core/src/test/scala/ml/combust/mleap/core/feature/NGramModelSpec.scala [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package ml.combust.mleap.core.feature
+
+import org.scalatest.FunSpec
+
+/**
+  * Created by mikhail on 10/16/16.
+  */
+class NGramModelSpec extends FunSpec{
+  describe("#apply") {
+    it("returns an array of ngrams") {
```
Comment (mishkinf) [inline mleap-core/src/test/scala/ml/combust/mleap/core/feature/NGramModelSpec.scala]: this is only testing the happy path. what about checking the error cases?

**Pull request merged**